### PR TITLE
Update + Compatibility with TTT2 + Code Cleaning

### PR DESCRIPTION
**Pull request by Alf21**
- Updated addon to make it compatible with TTT (https://steamcommunity.com/sharedfiles/filedetails/?id=404599106)
- Added Compatibility with TTT2
- Code Cleanup

### CODE REVIEW

**Comment by Alf21** [comment]
I implemented your last commits. There were some differences between this version and the workshop version I used. It should be compatible (able to merge) now...

**Comment by MinIsMin** [comment]
Looks good so far.
The only change for TTT2 compatibility seems to be the `if not ROLES then` part in [lua/cl_specdm_hud.lua](https://github.com/Tommy228/TTT_Spectator_Deathmatch/pull/73/files#diff-a83c59e89104f3c45ff22ead8b0ff4b3R44), or am I missing something?

What do you mean by 

> Updated addon to make it compatible with TTT (https://steamcommunity.com/sharedfiles/filedetails/?id=404599106)

?

The workshop version is going to be updated to master when [v1.10](https://github.com/Tommy228/TTT_Spectator_Deathmatch/milestone/5) is released.

**Comment by Alf21** [comment]
Year nearly (with some small modifications like adding colors of TTT2).

Ah alright. I thought that the steam workshop version is a newer one. But that was wrong, it was a older one, so i needed to modify it with the second commit

File: lua/cl_specdm_hud.lua [hunk 1]
```diff
@@ -29,106 +29,145 @@ local ammo_colors = {
...
 	end
 end
 
 local function GetAmmo(ply)
 	local weap = ply:GetActiveWeapon()
-	if not weap or not ply:Alive() then return -1 end
+    
+	if not weap or not ply:Alive() then 
+        return -1 
+    end
```
Comment (MinIsMin) [inline lua/cl_specdm_hud.lua]: What is this indentation? Shouldn't it be more to the right?
Many of your changed [`if`](https://github.com/Tommy228/TTT_Spectator_Deathmatch/pull/73/files#diff-5b6478f543c044b73e49d5235bf3e5c5R128) and [`timer`](https://github.com/Tommy228/TTT_Spectator_Deathmatch/pull/73/files#diff-dfff33395cd2f31d3eca5431f44bd6efR96) blocks have strange indentation.

If it's not intended I can change it, just wanted to ask before.

Comment (Alf21) [reply lua/cl_specdm_hud.lua]: hmmm, that's strange... I use auto intent, but it looks like that there are some spaces... Definitely not wanted

Comment (Alf21) [reply lua/cl_specdm_hud.lua]: So feel free to change it :)

**Comment by MinIsMin** [comment]
I will try to come up with a solution how to handle the spaces later.
(I tend to using spaces instead of tabs)

**Comment by Alf21** [comment]
Alright. Sorry. I thought you use tabs commonly. So I replaced every 4-time Spaces with 1 tab to avoid different styles.

File: lua/cl_specdm_hud.lua [hunk 2]
```diff
@@ -29,106 +29,145 @@ local ammo_colors = {
...
-	end
+	else
+        if not ROLES then
+            if client:GetTraitor() then
+                col = bg_colors.traitor
+            elseif client:GetDetective() then
+                col = bg_colors.detective
+            end
+        else
+            col = client.GetRoleData().color
```
[the remark this answers is not in the document]
Comment (MinIsMin) [reply lua/cl_specdm_hud.lua]: Next release (which will be also pushed to Steam Workshop) is set for this evening, if you want that small change in it, push a small PR or if it's only from a dot to colon I can change it.

**Pull request merged**